### ISSUE TITLE
Getting NotFoundError: when trying out structured data regressor example

### ISSUE THREAD

**Issue by sibyjackgrove** [keras-team/autokeras#1106]
### Bug Description
Getting the following error when trying out the structured data regressor example:
```
NotFoundError: Failed to create a NewWriteableFile: .\structured_data_regressor\trial_718a34f7ece7dd95949132eb383d4385\checkpoints\epoch_0\checkpoint_temp_eb81c6391cc7417681ea65e7f0049724/part-00000-of-00001.data-00000-of-00001.tempstate17467846151752881921 : The system cannot find the path specified.
; No such process [Op:SaveV2]
```
### Bug Reproduction
```
column_names = ['MPG','Cylinders','Displacement','Horsepower','Weight',
                'Acceleration', 'Model Year', 'Origin']
dataset_path = "http://archive.ics.uci.edu/ml/machine-learning-databases/auto-mpg/auto-mpg.data"
raw_dataset = pd.read_csv(dataset_path, names=column_names,
                      na_values = "?", comment='\t',
                      sep=" ", skipinitialspace=True)

dataset = raw_dataset.copy()
dataset = dataset.dropna()
train_dataset = dataset.sample(frac=0.8,random_state=0)
test_dataset = dataset.drop(train_dataset.index)

data_type = (len(column_names)-1) * ['numerical'] + ['categorical']
data_type = dict(zip(column_names, data_type))
column_names.remove('MPG')
del data_type['MPG']
regressor = ak.StructuredDataRegressor(max_trials=2, column_names=column_names, column_types=data_type)
regressor.fit(x=train_dataset.drop(columns=['MPG']), y=train_dataset['MPG'])
```
Data used by the code:
`dataset_path = "http://archive.ics.uci.edu/ml/machine-learning-databases/auto-mpg/auto-mpg.data"`
### Expected Behavior
Expected to train the model.

### Setup Details
Include the details about the versions of:
 - OS type and version: Windows 10
 - Python: 3.7.4
 - autokeras: master
 - keras-tuner: 1.0.1
 - scikit-learn: '0.21.3
 - numpy:1.18.3
 - pandas:0.24.2
 - tensorflow: 2.2.0-dev20200422

**Comment by psonglao** [comment]
Try specifying a shallow folder for the directory, e.g.
`regressor = ak.StructuredDataRegressor(max_trials=2, column_names=column_names, column_types=data_type, directory=r'C:\hello')`

I experienced a similar issue for StructuredDataClassifier. I think this error is happening because of a Windows limitation on maximum path length. Solved it by changing the directory to a very shallow folder as suggested here: https://github.com/una-dinosauria/human-motion-prediction/issues/10
